x/staking/keeper/msg_server.go

```diff
@@ -210,18 +203,35 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
 		)
 	}
 
+	tokens := msg.Amount.Amount
+	shares, err := validator.SharesFromTokens(tokens)
```

Comment:
Shouldn't this be checked and an error returned only if the delegator is a liquid staker?

Reply:
Thanks for such a quick reaction.

I have a question though, if you don't mind?

I found out when testing that there can be validators with 0 shares in some edge cases. In those edge cases `shares, err := validator.SharesFromTokens(tokens)` would raise an error with the message: `ErrInsufficientShares (insufficient delegation shares)`.

Do you think such a situation could occur in production? Asking because the change you made stops the tests from complaining, so I'm just double checking 😄

Reply:
I think it technically is possible if you undelegate everything from a validator and then attempt to delegate again while the unbonding is in progress. But I think an error should be thrown in this scenario anyways